src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -2017,6 +2017,72 @@ string[] linkingTargetFields
+        [DataRow("test-configs/my-config.json", "Test", true, false, "test-configs/dab-config.Test.json", "test-configs/my-config.json", DisplayName = "Config file in different directory provided by user and environment variable is set and environment file is present")]
+        [DataRow("my-config.json", "", false, true, null, "my-config.json", DisplayName = "Config file in the current directory provided by user and environment variable is not set and absolute path is provided")]
+        [DataRow("test-configs/my-config.json", "", true, true, null, "test-configs/my-config.json", DisplayName = "Config file in different directory provided by user and environment variable is not set and absolute path is provided")]
+        [DataRow("my-config.json", "Test", false, true, "my-config.Test.json", "my-config.json", DisplayName = "Config file in the current directory provided by user and environment variable is set and absolute path is provided")]
+        [DataRow("test-configs/my-config.json", "Test", true, true, "test-configs/my-config.Test.json", "test-configs/my-config.json", DisplayName = "Config file in different directory provided by user and environment variable is set and absolute path is provided")]
+        [DataRow("my-config.json", "Test", false, true, "dab-config.Test.json", "my-config.json", DisplayName = "Config file in the current directory provided by user and environment variable is set and environment file is present and absolute path is provided")]
+        [DataRow("test-configs/my-config.json", "Test", true, true, "test-configs/dab-config.Test.json", "test-configs/my-config.json", DisplayName = "Config file in the different directory provided by user and environment variable is set and environment file is present and absolute path is provided")]
+        [DataRow(null, "", false, false, null, "dab-config.json", DisplayName = "Config file not provided by user and environment variable is not set")]
+        [DataRow(null, "Test", false, false, "dab-config.Test.json", "dab-config.json", DisplayName = "Config file not provided by user and environment variable is set and environment file is present")]
+        [DataRow(null, "Test", false, false, null, "dab-config.json", DisplayName = "Config file not provided by user and environment variable is set and environment file is not present")]
+        public void TestCorrectConfigFileIsSelectedForRuntimeEngine(
+            string userProvidedConfigFilePath,
+            string environmentValue,
+            bool IsDifferentDirectory,
+            bool isAbsolutePath,
+            string environmentFile,
+            string finalConfigFilePath)
+        {
+            MockFileSystem fileSystem = new();
+            if (IsDifferentDirectory)
+            {
+                fileSystem.AddDirectory("test-configs");
+            }
+
+            if (isAbsolutePath)
```

Comment:
Would make test easier to read if `isAbsolutePath` was not an additional test parameter. (Just lots of booleans to read and parse).
You could instead use [https://learn.microsoft.com/en-us/dotnet/api/system.io.path.ispathrooted?view=net-6.0#system-io-path-ispathrooted(system-string)](`Path.IsPathRooted`)

Reply:
I'm using this variable not to know if the provided path is Absolute path or not, but to make it absolute, since it's a MockFileSystem, we need to make the path absolute programmatically.

Reply:
updated the variable name to `useAbsolutePath` to indicate it's correct usage